src/controllers/settings.js

```diff
@@ -2,16 +2,191 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import {
+  getUserEmails,
+  resetUnverifiedEmailAddress,
+  addSubscriberUnverifiedEmailHash,
+  removeOneSecondaryEmail,
+  getEmailById,
+  verifyEmailHash
+} from '../db/tables/email_addresses.js'
+
+import { setAllEmailsToPrimary } from '../db/tables/subscribers.js'
+
+import { fluentError, getMessage } from '../utils/fluent.js'
+import { sendEmail, getVerificationUrl, getUnsubscribeUrl } from '../utils/email.js'
 import { mainLayout } from '../views/main.js'
 import { settings } from '../views/partials/settings.js'
+import AppConstants from '../app-constants.js'
+import { getBreachesForEmail } from '../utils/hibp.js'
+import { generateToken } from '../utils/csrf.js'
+
+import { getTemplate } from '../emails/templates/email-2022.js'
+import { verifyPartial } from '../emails/partials/email-verify.js'
+
+async function settingsPage (req, res) {
+  const emails = await getUserEmails(req.session.user.id)
+  // Add primary subscriber email to the list
+  emails.push({
+    email: req.session.user.primary_email,
+    sha1: req.session.user.primary_sha1,
+    primary: true,
+    verified: true
+  })
+
+  const breachCounts = new Map()
+  emails.forEach((email) => breachCounts.set(email.email, 0))
+
+  const allBreaches = req.app.locals.breaches
+  for (const email of emails) {
+    const breaches = await getBreachesForEmail(
+      email.sha1,
+      allBreaches,
+      true,
+      false
+    )
+    breachCounts.set(email.email, breaches.length)
```

Comment:
```suggestion
    breachCounts.set(email.email, breaches?.length || 0)
```
And avoid having 2 loops (remove ln 38)?